client/ayon_core/pipeline/plugin_discover.py

```diff
@@ -138,7 +138,14 @@ def discover_plugins(
         for item in modules:
             filepath, module = item
             result.add_module(module)
-            all_plugins.extend(classes_from_module(base_class, module))
+            for cls in classes_from_module(base_class, module):
+                if cls is base_class:
+                    continue
+                # Class has defined 'is_base_class = True'
+                is_base_class = cls.__dict__.get("is_base_class")
```

Comment:
Result should always be a bool ?
```suggestion
                is_base_class = cls.__dict__.get("is_base_class", False)
```